extensions/arrow/src/main/java/io/deephaven/extensions/arrow/sources/ArrowInstantColumnSource.java

```diff
@@ -48,6 +73,7 @@ public final Instant get(final long rowKey) {
     }
 
     private Instant extract(final int posInBlock, final TimeStampVector vector) {
-        return vector.isSet(posInBlock) == 0 ? null : io.deephaven.time.DateTimeUtils.epochNanosToInstant(vector.get(posInBlock));
+        final long value = vector.get(posInBlock);
+        return vector.isSet(posInBlock) == 0 ? null : DateTimeUtils.epochNanosToInstant(value * factor);
```

Comment:
This is eagerly reading vector.get; as far as I know, this sort of "always access the value" pattern may be better, but I will note it is different than before where it only happened when the value was set. Happy to approve, just wanted to make sure we don't think it matters.

Reply:
Oh my gosh, thank you for noticing... when testing the barrage types change, I discovered that Vector impls are very very picky about `#get`ing unset values. Noting that `#getObject` is able to return null, whereas `#get` tends to be primitive.

```java
    public long get(int index) throws IllegalStateException {
        if (NullCheckingForGet.NULL_CHECKING_ENABLED && this.isSet(index) == 0) {
            throw new IllegalStateException("Value at index is null");
        } else {
            return this.valueBuffer.getLong((long)index * 8L);
        }
    }
 ```
 
 Definitely the case here. The test file DHE provided must not have any null values.